controllers/transactions.js

```diff
@@ -0,0 +1,36 @@
+const transactionsModel = require('../models/transaction')
+const { getUserId } = require('../utils/users')
+
+/**
+ * Collects all transactions and sends only required data for username spicified in url
+ * @param req {Object} - Express request object
+ * @param res {Object} - Express response object
+ */
+
+const fetch = async (req, res) => {
+  try {
+    const userId = await getUserId(req.params.username)
+    if (userId) {
+      const n = req.query.n || 10
+      const o = req.query.o || 'DESC'
+      const data = await transactionsModel.fetch(userId, n, o)
+      if (data.length > 0) {
+        return res.json({
+          message: 'Transactions returned successfully!',
+          data
+        })
+      } else {
+        return res.boom.notFound('No transactions exist!')
+      }
```

Comment:
Transactions not existing is not the user's problem. 4xx response is when there's something wrong from the user's side.
The API call isn't bad when there are no transactions. It should be 200 with appropriate message.

Have a look here;
https://github.com/Real-Dev-Squad/website-backend/blob/7bc36f4efe28d434f9f55bcbdab10c66bb1f7aed/controllers/members.js#L16

Reply:
Done